lib/mock/mock-interceptor.js

```diff
@@ -99,43 +99,40 @@ class MockInterceptor {
     return { statusCode, data, headers, trailers }
   }
 
-  validateReplyParameters (statusCode, data, responseOptions) {
-    if (typeof statusCode === 'undefined') {
+  validateReplyParameters (replyParameters) {
+    if (typeof replyParameters.statusCode === 'undefined') {
       throw new InvalidArgumentError('statusCode must be defined')
     }
-    if (typeof data === 'undefined') {
-      throw new InvalidArgumentError('data must be defined')
-    }
-    if (typeof responseOptions !== 'object' || responseOptions === null) {
+    if (typeof replyParameters.responseOptions !== 'object' || replyParameters.responseOptions === null) {
       throw new InvalidArgumentError('responseOptions must be an object')
     }
   }
 
   /**
    * Mock an undici request with a defined reply.
    */
-  reply (replyData) {
+  reply (replyOptionsCallbackOrStatusCode) {
     // Values of reply aren't available right now as they
     // can only be available when the reply callback is invoked.
-    if (typeof replyData === 'function') {
+    if (typeof replyOptionsCallbackOrStatusCode === 'function') {
       // We'll first wrap the provided callback in another function,
       // this function will properly resolve the data from the callback
       // when invoked.
       const wrappedDefaultsCallback = (opts) => {
         // Our reply options callback contains the parameter for statusCode, data and options.
-        const resolvedData = replyData(opts)
+        const resolvedData = replyOptionsCallbackOrStatusCode(opts)
 
         // Check if it is in the right format
-        if (typeof resolvedData !== 'object') {
+        if (typeof resolvedData !== 'object' || resolvedData === null) {
```

Comment:
if we dont do this, than null would result in destructuring issues in the original code